SAP/fundamental-ngx
# test(core, platform): unit test fixes without ivy enabled

#### Please provide a link to the associated issue.

Closes #5742, #4206, #4208, #4210, #4196.

#### Please provide a brief summary of this pull request.

Unit tests improvements, error fixes.

To test:
* Add `"angularCompilerOptions": { "enableIvy": false }` to the `tsconfig.base.json`.
* Run `npm run test` & `npm run test:platform`
* See working tests :)

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [n/a] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [n/a] Documentation Examples
- [n/a] Stackblitz works for all examples
